### DISCUSSION TITLE
Updating parent entity's `updatedTime` field when updating child entity in one-to-many relationship?

### DISCUSSION THREAD

**Discussion by kevinwhkim** [prisma/prisma#10420] [Q&A]
Let's say there are two entities in a one-to-many relationship, `Parent` and `Child`, with the schema looking like this:
```
datasource db {
  provider = "postgresql"
  url = env("DATABASE_URL")
}

model Parent {
    uuid  String  @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
    name  String  @map(name: "name") @db.VarChar(20)
    age  Int  @default(0) @db.SmallInt
    createdTime  DateTime  @default(now()) @map(name: "created_time") @db.Timestamptz(6)
    updatedTime  DateTime  @default(now()) @updatedAt @map(name: "updated_time") @db.Timestamptz(6)
    children  Child[]

    @@map("parent")
}

model Child {
    uuid  String  @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
    name  String  @map(name: "name") @db.VarChar(20)
    age  Int  @default(0) @db.SmallInt
    createdTime  DateTime  @default(now()) @map(name: "created_time") @db.Timestamptz(6)
    updatedTime  DateTime  @default(now()) @updatedAt @map(name: "updated_time") @db.Timestamptz(6)
    parentUuid  String?  @map(name: "parent_uuid")
    parent  Parent?  @relation(fields: [parentUuid], references: [uuid])
   
    @@map("child)
}
```

I want it so that when I update any parent's children (or create new children relating to a parent) I automatically set the parent's `updatedTime` value to the current time, using an example nested query below:

```
prisma.parent.update({
    where: {
        uuid: someParent.uuid,
    },
    data: {
        children: {
            create: { name: "Joe", age: 30 }
        }
    }
});
```

How would I achieve this? I am aware of `@updateAt` but that only updates if you've updated fields for the parent and not if you've only created/updated its children. I've also looked into using middleware but just wanted to ask if there is a simpler solution that I may have overlooked.

**Comment by vasilii-kovalev** [comment]
That would also be useful for many-to-many relationship too. Here's an example of a partial user's update.

```typescript
const user = await prismaClient.user.update({
  data: {
    email: !isUndefined(email)
      ? email.toLowerCase()
      : undefined,
    name,
    password: !isUndefined(password)
      ? await hashUserPassword(password)
      : undefined,
    roles: !isUndefined(roles)
      ? {
        set: roles.map((roleId) => {
          return {
            id: roleId,
          };
        }),
      }
      : undefined,
  },
  select: USER_SELECTOR,
  where: {
    id,
  },
});
```

If only `roles` field is updated, the `user` model's field marked with `@updatedAt` doesn't change automatically.

Currently, to fix this, it is necessary to manually update the marked field if the relations were updated:

```diff
const user = await prismaClient.user.update({
  data: {
    email: !isUndefined(email)
      ? email.toLowerCase()
      : undefined,
    name,
    password: !isUndefined(password)
      ? await hashUserPassword(password)
      : undefined,
    roles: !isUndefined(roles)
      ? {
        set: roles.map((roleId) => {
          return {
            id: roleId,
          };
        }),
      }
      : undefined,
+   updatedDate: !isUndefined(roles)
+     ? new Date()
+     : undefined,
  },
  select: USER_SELECTOR,
  where: {
    id,
  },
});
```

But it is suboptimal, because
1. There may be several updating relations, and it is necessary to check all of them
2. There may be several places like this throughout the code base, and it is necessary to manually check that every place has this logic

**Reply by vasilii-kovalev** [reply]
I've also noticed, that after changing `roles`, the field marked with `@updatedAt` in the records of the `role` table doesn't change as well.
I've tried to add `updatedDate: new Date()` alongside with `id: roleId` in the `roles.map`, but it leads to the following error:

> An operation failed because it depends on one or more records that were required but not found. Expected 1 records to be connected, found only 0.

Adding `@default(now())` to the field's definition in the Prisma schema in hope that the field will be automatically updated somehow doesn't help, so it seems for this problem there is no workaround.

**Reply by ludralph** [reply]
We do have an open [issue](https://github.com/prisma/prisma/issues/17274) to address this. I also noticed you already commented there.  Sadly, there isn't a workaround for this yet.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
